### PR TITLE
integ/tests: Reduce transient failures

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
A test or two will fail randomly during integration tests (though every other test passes)
The command `sam local start-api` appears to sometimes build the image twice for the 'image' + 'start-api' combo, causing the test to fail due to timeout.

## Solution
* Add a start-up delay before running tests (20 seconds for now)
* Force 'skip-pull-image' when invoking an API lambda.
* Small refactor to `codelens.js.test.ts` (why is this named like that) replacing `expectCodeLenses` with `getAddConfigCodeLens`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]

File: src/integrationTest/sam.test.ts [hunk 1]
```diff
@@ -233,7 +233,12 @@ async function getAddConfigCodeLens(documentUri: vscode.Uri): Promise<vscode.Cod
     return waitUntil(
         async () => {
             try {
-                let codeLenses = await getCodeLenses(documentUri)
+                // Sometimes 'getCodeLenses' will never return, so we need to race it
+                let codeLenses = await Promise.race<Promise<vscode.CodeLens[] | undefined>>([
```
Comment (bryceitoc9) [inline src/integrationTest/sam.test.ts:237]: Does this change make the old CODELENS_RETRY_INTERVAL usage (now line 264) redundant?

Comment (JadenSimon) [reply src/integrationTest/sam.test.ts:237]: Good question. The 2nd retry interval is used for when `getCodeLens` throws an error before the other Promise resolves. The code would actually now wait a total of 2*`CODELENS_RETRY_INTERVAL` if `getCodeLens` stalls.

**Comment by JadenSimon** [comment]
The patches fixed nodejs10 failing, but now other CodeLens tests will randomly fail. Will come back to this later.

File: src/shared/sam/localLambdaRunner.ts [hunk 2]
```diff
@@ -228,7 +228,7 @@ async function invokeLambdaHandler(
             debugPort: debugPort,
             debuggerPath: config.debuggerPath,
             debugArgs: config.debugArgs,
-            skipPullImage: config.sam?.skipNewImageCheck,
+            skipPullImage: true, // We already built the image, but `sam local start-api` will try to build it again
```
Comment (justinmk3) [inline src/shared/sam/localLambdaRunner.ts:231]: good idea. should we also do this for the `SamCliLocalInvokeInvocationArguments` case below?

Comment (JadenSimon) [reply src/shared/sam/localLambdaRunner.ts:231]: Yeah makes sense to me. Still can't figure out why the very first CodeLens fails though. Strange.

**Review by justinmk3 — APPROVED** [review]

**Comment by JadenSimon** [comment]
Ran 3 times with no failures. Looks good to merge.

**Pull request merged**